lib/iphone-simulator-xcode-simctl.ts

```diff
@@ -116,11 +120,63 @@ export class XCodeSimctlSimulator extends IPhoneSimulatorNameGetter implements I
 	}
 
 	public printDeviceLog(deviceId: string, launchResult?: string): any {
-		return common.printDeviceLog(deviceId, launchResult);
+		let pid = "";
+		let deviceLogChildProcess;
+
+		if (launchResult) {
+			pid = launchResult.split(":")[1].trim();
+		}
+
+		if (!this.isDeviceLogOperationStarted) {
+			deviceLogChildProcess = this.getDeviceLogProcess(deviceId);
+			if (deviceLogChildProcess.stdout) {
+				deviceLogChildProcess.stdout.on("data", (data: NodeBuffer) => {
+					let dataAsString = data.toString();
+					if (pid) {
+						if (dataAsString.indexOf(`[${pid}]`) > -1 || dataAsString.indexOf(` ${pid} `) > -1) {
+							process.stdout.write(dataAsString);
+						}
+					} else {
+						process.stdout.write(dataAsString);
+					}
+				});
+			}
+
+			if (deviceLogChildProcess.stderr) {
+				deviceLogChildProcess.stderr.on("data", (data: string) => {
+					let dataAsString = data.toString();
+					if (pid) {
+						if (dataAsString.indexOf(`[${pid}]`) > -1 || dataAsString.indexOf(` ${pid} `) > -1) {
+							process.stdout.write(dataAsString);
+						}
+					} else {
+						process.stdout.write(dataAsString);
+					}
+					process.stdout.write(data.toString());
+				});
+			}
+		}
+
+		return deviceLogChildProcess;
 	}
 
 	public getDeviceLogProcess(deviceId: string): any {
-		return common.getDeviceLogProcess(deviceId);
+		const device = this.getDeviceFromIdentifier(deviceId) || {};
```

Comment:
The initialization of these variables can be done inside the `if` statement down below, because they're only used there